nifi-nar-bundles/nifi-deeplearning4j-bundle/nifi-deeplearning4j-processors/src/main/java/org/apache/nifi/processors/deeplearning4j/AbstractDeepLearning4JProcessor.java

```diff
@@ -0,0 +1,106 @@
+            .displayName("Character Set")
+            .description("Specifies the character set of the document data.")
+            .required(true)
+            .defaultValue("UTF-8")
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .addValidator(StandardValidators.CHARACTER_SET_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor FIELD_SEPARATOR = new PropertyDescriptor.Builder()
+            .name("deeplearning4j-field-separator")
+            .displayName("Field Separator")
+            .description("Specifies the field separator in the records. (default is comma)")
+            .required(true)
+            .defaultValue(",")
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor RECORD_SEPARATOR = new PropertyDescriptor.Builder()
+            .name("deeplearning4j-record-separator")
+            .displayName("Record Separator")
+            .description("Specifies the records separator in the message body. (defaults to new line)")
+            .required(true)
+            .defaultValue(System.lineSeparator())
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor MODEL_FILE = new PropertyDescriptor.Builder()
+            .name("model-file")
+            .displayName("Model File")
+            .description("Location of the Deeplearning4J model zip file")
+            .required(true)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .addValidator(StandardValidators.FILE_EXISTS_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor RECORD_DIMENSIONS = new PropertyDescriptor.Builder()
+            .name("deeplearning4j-record-dimension")
+            .displayName("Record dimensions separated by field separator")
+            .description("Dimension of array in each a record (eg: 2,4 - a 2x4 array)")
+            .required(true)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final String DEEPLEARNING4J_ERROR_MESSAGE = "deeplearning4j.error.message";
+
+    public static final String DEEPLEARNING4J_OUTPUT_SHAPE = "deeplearning4j.output.shape";
+
+    protected MultiLayerNetwork model = null;
+
+    protected synchronized MultiLayerNetwork getModel(ProcessContext context) throws IOException {
+        if ( model == null ) {
+            String modelFile = context.getProperty(MODEL_FILE).evaluateAttributeExpressions().getValue();
+            if ( getLogger().isDebugEnabled()) {
```

Comment:
Probably not necessary to bother checking for isDebugEnabled

Reply:
Corrected.